security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -519,8 +580,63 @@ func (sc *SecretController) refreshSecret(scrt *v1.Secret) error {
 
 	scrt.Data[CertChainID] = chain
 	scrt.Data[PrivateKeyID] = key
-	scrt.Data[RootCertID] = sc.ca.GetCAKeyCertBundle().GetRootCertPem()
+	scrt.Data[RootCertID] = sc.getTrustAnchorsBundle()
 
 	_, err = sc.core.Secrets(namespace).Update(scrt)
 	return err
 }
+
+// The caller is responsible for holding the sc.extraTrustAnchorMu lock while calling this method.
+func (sc *SecretController) refreshTrustAnchorsBundle() {
+	trustAnchorsSet := map[string]struct{}{}
+
+	sc.primaryTrustAnchor = sc.ca.GetCAKeyCertBundle().GetRootCertPem()
+	primary := strings.TrimSpace(string(sc.primaryTrustAnchor))
+	trustAnchorsSet[primary] = struct{}{}
+
+	for _, configMap := range sc.extraTrustAnchors {
+		for _, value := range configMap {
+			cert := strings.TrimSpace(value)
+			trustAnchorsSet[cert] = struct{}{}
+		}
+	}
+
+	trustAnchors := make([]string, 0, len(trustAnchorsSet))
+	for trustAnchor := range trustAnchorsSet {
+		trustAnchors = append(trustAnchors, trustAnchor)
+	}
+	sort.Strings(trustAnchors)
+
+	sc.combinedTrustAnchorsBundle = []byte(strings.Join(trustAnchors, "\n"))
```

Comment:
I _think_ we also want to update the `caTLSRootCert` in `istio-security` configmap to include the extra trusted roots for this to work with SDS. NodeAgent might also need to start watching for changes. @myidpt, what do you think?

Reply:
This change would be needed if we decided to use the trust anchors for cert rotation.

Reply:
I think whatever is done for file mount has to be done for SDS. @myidpt WDTY?

Reply:
How extra trusted roots are added (configmap) should be orthogonal to how the resulting bundle is distributed (file vs. SDS).

Reply:
SDS doesn't support multiple trust anchors :(
https://github.com/istio/istio/blob/master/security/pkg/server/ca/server.go#L109
https://github.com/istio/istio/blob/master/security/proto/istioca.proto#L33
The proto we are using assumes that the root is only the last cert in the cert chain, so it doesn't have an extra field to specify multiple roots / trust anchors.

Besides that, just to this question: the `caTLSRootCert` in `istio-security` configmap doesn't need to be changed. It's only needed for Node Agent to authenticate to Citadel.

Reply:
> The caTLSRootCert in istio-security configmap doesn't need to be changed. It's only needed for Node Agent to authenticate to Citadel.

Sharing the current root CA in a configmap is very useful. Tooling doesn't need access to secrets or private keys to read the public cert.